src/Shell/UpgradeShell.php

```diff
@@ -75,7 +75,8 @@ public function all()
             if (!empty($this->params['interactive'])) {
                 $continue = $this->in('Continue with `' . $action . '`?', ['y', 'n', 'q'], 'y');
                 if ($continue === 'q') {
-                    return $this->error('Aborted. Changes are not commited.');
+                    $this->abort('Aborted. Changes are not commited.');
+                    return;
```

Comment:
the return is not necessary then

Reply:
`return` is unreachable, but it is easier for static code analysis to understand where function exits.

Reply:
Are you sure? This shouldnt make any difference, and we never added those before.

Reply:
I thought `abort()` uses `exit()`, but now I see it uses exception.

If it would use `exit`, I still think `return` could.be useful in [mocking](https://github.com/cakephp/cakephp/pull/10044#discussion_r96154294), but now with exceptions it is easier to test.